x-pack/plugins/rule_registry/server/utils/create_persistence_rule_type_wrapper.ts

```diff
@@ -85,13 +85,25 @@ export const createPersistenceRuleTypeWrapper: CreatePersistenceRuleTypeWrapper
                   return { createdAlerts: [], errors: {}, alertsWereTruncated: false };
                 }
 
+                let enrichedAlerts = filteredAlerts;
```

Comment:
IMO, the enrichments should occur before alerts are passed to `alertWithPersistence`. Apologies if this decision to put it in the rule registry was already discussed, as perhaps there's context I'm not aware of. Since the alerting framework plans to eventually absorb the rule registry (and eventually, this function), the alerts that are passed in should be ready to go.

cc @mikecote, @kobelb

Reply:
Hey, it was done for performance reasons.

To be able to utilise deduplication which happens above in this function.

Otherwise, it can happen that we try to enrich alerts which will be not written later into the alert index. It can lead to increased time for rule execution and just unnecessary enrichment request.

Reply:
Resolved in https://github.com/elastic/kibana/pull/139478#pullrequestreview-1106521698.